src/oidcc_token.erl

```diff
@@ -349,6 +356,130 @@ retrieve(AuthCode, ClientContext, Opts) ->
             {error, {grant_type_not_supported, authorization_code}}
     end.
 
+%% @doc
+%% Validate the JARM response, returning the valid claims as a map.
+%%
+%% The response was sent to the local endpoint by the OpenId Connect provider,
+%% using redirects
+%%
+%% <h2>Examples</h2>
+%%
+%% ```
+%% {ok, ClientContext} =
+%%   oidcc_client_context:from_configuration_worker(provider_name,
+%%                                                  <<"client_id">>,
+%%                                                  <<"client_secret">>),
+%%
+%% %% Get Response from Redirect
+%%
+%% {ok, #{<<"code">> := AuthCode}} =
+%%   oidcc:validate_jarm(Response, ClientContext, #{}),
+%%
+%% {ok, #oidcc_token{}} = oidcc:retrieve(AuthCode, ClientContext,
+%%   #{redirect_uri => <<"https://redirect.example/">>}.
+%% '''
+%% @end
+%% @since 3.2.0
+-spec validate_jarm(Response, ClientContext, Opts) ->
+    {ok, oidcc_jwt_util:claims()} | {error, error()}
+when
+    Response :: binary(),
+    ClientContext :: oidcc_client_context:t(),
+    Opts :: validate_jarm_opts().
+validate_jarm(Response, ClientContext, Opts) ->
+    #oidcc_client_context{
+        provider_configuration = Configuration,
+        client_id = ClientId,
+        client_secret = ClientSecret,
+        client_jwks = ClientJwks,
+        jwks = Jwks
+    } = ClientContext,
+    #oidcc_provider_configuration{
+        issuer = Issuer,
+        response_modes_supported = ResponseModesSupported,
+        authorization_signing_alg_values_supported = SigningAlgSupported0,
+        authorization_encryption_alg_values_supported = EncryptionAlgSupported0,
+        authorization_encryption_enc_values_supported = EncryptionEncSupported0
+    } =
+        Configuration,
+
+    case
```

Comment:
Do we need to validate the response modes when the user already has a JARM? That seems more like a thing that we should validate when requesting the token / processing that response.

If the user gets the JARM by some custom extension, it should still be able to validate.

Reply:
Yep, and that makes the code much simpler. I've removed all of this.